### PR TITLE
Bump entire site, except heroicons and router

### PR DESCRIPTION
**Pull request by birkskyum**
Similar to https://github.com/solidjs/solid-playground/pull/165, gets things up to speed.

This runs best if deployed on Node 20, due to packages like [sindresorhus/got](https://github.com/sindresorhus/got/releases), Vite 5 (18/20+) etc.

Router requires refactor - is best to do in follow-up PR (#531 )

Also closes:
- #482
- #484
- #511
- #512

@davedbase , maybe you can have a look at this one?

### CODE REVIEW

**Comment by birkskyum** [comment]
Any chance for a review on this before it gets obsoleted by other PRs ?

File: src/AppContext.tsx
Comment (atk) [inline src/AppContext.tsx:1]: This looks like there are merely formatting changes in this file. Maybe revert this chunk?

**Review by atk — APPROVED** [review]
Looks good to me

File: src/AppContext.tsx
Comment (birkskyum) [reply src/AppContext.tsx:1]: Yep, I've reset that file.

File: postcss.config.cjs [hunk 1]
```diff
@@ -1,3 +1,6 @@
 module.exports = {
-  plugins: [require('tailwindcss'), require('autoprefixer')],
+  plugins: [
+    require('tailwindcss/nesting'),
```
Comment (modderme123) [inline postcss.config.cjs:3]: What do we use nesting for?

Comment (birkskyum) [reply postcss.config.cjs:3]: I got this build message, which was resolved by adding the `tailwindcss/nesting`

```sh
[vite:css] Nested CSS was detected, but CSS nesting has not been configured correctly.
Please enable a CSS nesting plugin *before* Tailwind in your configuration.
See how here: https://tailwindcss.com/docs/using-with-preprocessors#nesting
8  |    overflow-y: hidden;
9  |  
10 |    .focused {
   |   ^^^^^^^^^^
11 |            --vscode-multiDiffEditor-border: var(--vscode-focusBorder);
   |  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
12 |    }
```

**Pull request merged**